client/engine/engine.go

```diff
@@ -152,26 +162,20 @@ func (e *Engine) handleMessage(message protocols.Message) (ObjectiveChangeEvent,
 // attempts progress.
 func (e *Engine) handleChainEvent(chainEvent chainservice.Event) (ObjectiveChangeEvent, error) {
 	e.logger.Printf("handling chain event %v", chainEvent)
-	objective, ok := e.store.GetObjectiveByChannelId(chainEvent.ChannelId)
+	objective, ok := e.store.GetObjectiveByChannelId(chainEvent.GetChannelId())
 	if !ok {
-		// Because the MockChain and SimpleChainService broadcast all events to all subscribers,
-		// it is likely that a client receives an irrelevant event. So we log and continue without
-		// returning an error.
-		e.logger.Printf("handleChainEvent: No objective in store for channel with id %s", chainEvent.ChannelId)
-		return ObjectiveChangeEvent{}, nil
+		return ObjectiveChangeEvent{}, &UnhandledChainEvent{event: chainEvent, objective: objective}
```

Comment:
This returns something like "chain event `x` could not be handled by objective nil", which is possibly less descriptive than we would like.

Reply:
maybe:

```
type UnhandledChainEvent struct {
	event     chainservice.Event
	objective protocols.Objective
	reason string
}

func (uce *UnhandledChainEvent) Error() string {
	return fmt.Sprintf("chain event could not be handled %#v by objective %#v: %s", uce.event, uce.objective, uce.reason)
}
```
which could be supplied things like "objective not found", "objective not a chainEventHandler", etc.

(ignores other suggestions)

Reply:
Good call! Done in 72b1b02